stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -185,8 +185,28 @@ end
     end
     r
 end
-diagzero(::Diagonal{T}, i, j) where {T} = zero(T)
-diagzero(D::Diagonal{<:AbstractMatrix{T}}, i, j) where {T} = zeros(T, size(D.diag[i], 1), size(D.diag[j], 2))
+"""
+    diagzero(A::AbstractMatrix, i, j)
+
+Return the appropriate zero element `A[i, j]` corresponding to a banded matrix `A`.
+"""
+diagzero(A::AbstractMatrix, i, j) = zero(eltype(A))
+diagzero(D::Diagonal{M}, i, j) where {T,M<:AbstractMatrix{T}} =
```

Comment:
```suggestion
diagzero(D::Diagonal{M}, i, j) where {M<:AbstractMatrix} =
```

You don't seem to use `T` anywhere?